### PR TITLE
Add sample: Front Door Premium (Preview) with App Service origin and Private Link

### PR DESCRIPTION
**Pull request by johndowns**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add sample: Front Door Premium (Preview) with App Service origin and Private Link

### CODE REVIEW

**Comment by johndowns** [comment]
Note that the ARM TTK is warning about the location parameter being hard-coded. This is intentional because during its preview, [Front Door restricts the regions that can be used for Private Link origins](https://docs.microsoft.com/en-us/azure/frontdoor/standard-premium/concept-private-link#limitations). Once this restriction is removed I plan to submit an update to this sample.

File: 201-front-door-premium-app-service-private-link/modules/app-service.bicep [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  location: location
+  kind: 'app'
+  properties: {
+    serverFarmId: appServicePlan.id
+  }
+}
+
+output appHostName string = app.properties.defaultHostName
+output appServiceResourceId string = app.id
+output appServiceLocation string = app.location
```
Comment (bmoore-msft) [inline 201-front-door-premium-app-service-private-link/modules/app-service.bicep:41]: this must be passed in so shouldn't be needed as an output

File: 201-front-door-premium-app-service-private-link/modules/app-service.bicep [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  name: appName
+  location: location
+  kind: 'app'
+  properties: {
+    serverFarmId: appServicePlan.id
+  }
+}
+
+output appHostName string = app.properties.defaultHostName
+output appServiceResourceId string = app.id
```
Comment (bmoore-msft) [inline 201-front-door-premium-app-service-private-link/modules/app-service.bicep:40]: this must be passed in so shouldn't be needed as an output

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-front-door-premium-app-service-private-link/modules/app-service.bicep (hunk 2 above)
Comment (johndowns) [reply 201-front-door-premium-app-service-private-link/modules/app-service.bicep:40]: I've removed the location, but resource ID is generated by the module so I've left that there.

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by johndowns** [comment]
@bmoore-msft Thanks - back to you for another review pass

**Pull request merged**